hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContract.php

```diff
@@ -193,21 +194,26 @@ protected static function getContractDates($contracts) {
     // If there are two (or more) Contracts starting on the same day
     // then we pick only the one with latest End date.
     foreach ($contracts['values'] as $contract) {
-      $details = civicrm_api3('HRJobDetails', 'getsingle', array(
+      $details = civicrm_api3('HRJobDetails', 'get', array(
```

Comment:
since we are inside job contract extension it is better to use BAO method to get the contract details ( it should be much faster too )